src/routes/MyPage.jsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { SmallPost } from "../components/Posts";
+import { MyPageInput } from "../components/EditInput";
+
+import { Link } from "react-router-dom";
+import { getPosts, getUser, getMyPage } from "../apis/api";
+import { getCookie } from "../utils/cookie";
+
+const MyPage = () => {
+  const [userProfile, setUserProfile] = useState();
+
+  const [tags, setTags] = useState([]);
+  const [searchTags, setSearchTags] = useState([]);
+  const [searchValue, setSearchValue] = useState("");
+  const [postList, setPostList] = useState([]);
+
+  useEffect(() => {
+    // access_token이 있으면 유저 정보 가져옴
+    if (getCookie("access_token")) {
+      const getMyPageAPI = async () => {
+        const userProfile = await getMyPage();
+        setUserProfile(userProfile);
```

Comment:
`userProfile`을 받아올 때 아마 major, college, user 이렇게 나눠져 있고, user 안에 username과 email이 있을 거예요. 굳이 이 상태로 쓸 필요는 없겠죠?
저라면 필요한 값들만 넣어줄 것 같습니다.

```suggestion
        setUserProfile({
        email : userProfile.user.email,
        username : userProfile.user.username,
        major : userProfile.major,
        college : userProfile.college,
        });
```

이런 식으로요!

그리고 userProfile 변수가 중복 사용된 것 같은데, 아마 문제는 없었을 것 같은데 그래도 최대한 변수명은 다르게 설정해주는게 좋아요!